app/main.py

```diff
@@ -2,5 +1,17 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def inner(*args, **kwargs) -> Any:
+        key = str(args) + str(kwargs)
```

Comment:
The key generation using `str(args) + str(kwargs)` may not be reliable for all types of arguments, especially if they are not hashable or if their string representation is not unique. Consider using a more robust method to generate keys, such as using a tuple of the arguments.